### PR TITLE
Initialize all the things

### PR DESCRIPTION
**Pull request by eliteraspberries**
In this branch I try to replace malloc with calloc as much as possible, to avoid uninitialized memory.

On Linux calloc returns immediately and the pages are zeroed when they are accessed, so the performance impact is minimal.

The changes build and all tests pass on OS X with Clang.

### CODE REVIEW

**Comment by mpg** [comment]
Thanks for your proposed patches! Overall they look very good. A couple of remarks though:

- I noticed there are uses of `polarssl_malloc()` remaining. Is that on purpose? If so, what's the rationale?
- Depending on the answer to the previous question, I'm wondering if we could/should stop using `polarssl_malloc()` completely, and rely on `polarssl_calloc()` only. This would seem more consistent, and slightly easier on users that want to override the function since there would be two, not three, memory-related functions. 
- In this vein, the memory_buffer_alloc implementation should be updated to provide calloc in addition or replacement to malloc.
- Unfortunately I'm not sure these patches are suitable for inclusion in the stable branch, since they introduce a new dependency.
- I really appreciate that you fixed the whitespace after each application of spatch. One minor nitpick however: we tend to use `sizeof( type )` with spaces. I'm not saying this is the best convention ever, but it's probably better to remain consistent.

Bottom line: we'll have some internal discussion and get back to you soon.

**Comment by eliteraspberries** [comment]
Yes, I'm hoping to switch all the remaining `polarssl_malloc` to `polarssl_calloc` too. I'll try to do as much as possible with Coccinelle, and the rest manually (about 20 left).

**Comment by mpg** [comment]
I didn't look in detail at your last patches but overall this looks good. However, as I mentioned before, if we integrate those changes, if will not be in the current stable branch, but in the next one. So I'd like to encourage you to base this line of patches on the (currently experimental) mbedtls-1.4 branch which will serve as a starting point for the next stable branch.

**Comment by mpg** [comment]
I applied these changes manually in our (currently internal) 2.0 branch, as they were not fit for the 1.3 branch as I mentioned above. I did most of the changes manually or with sed, but used you rm-zero-after-alloc semantic patch. Thanks for your help!

All the things are now initialized :) CVE-2015-1182 will not have any little brother.
